### PR TITLE
[FIX] project_timesheet_holidays: move init from company to post_init

### PR DESCRIPTION
**Pull request by dbkosky**
tldr: The init function should not be used in this case for creating records. Doing so results in errors. The content of this init function can be safely moved to the post-init hook

The problem:
In saas~17.2, creating a company with a country can result in the installation of its accompanying localisation. If it is a European country, this will trigger the installation of base_vat too. If project_timesheet_holdiays is installed this can result in the installation of base_vat failing.

The init method described on res_company in project_timesheet_holidays is triggered when the res_partner model is extended, for instance, in base_vat when adding fields. The registry will contain these fields, but the accompanying fields will not yet have been written to the db.

When the project and task are created, a chain of dependencies result in the company's partner being read. In turn, a query is issued using fields from the registry that aren't yet written to the DB, resulting in an SQL error.

Solution:
Move the content of this method to the post-init hook. This is safe to do and works the same way.

task-id: 4060907

### CODE REVIEW

**Review by xavierbol — COMMENTED** [review]
LGTM, thanks for your work. 🙂 

robodoo r+
